src/configuration.ts

```diff
@@ -120,8 +120,80 @@ export function loadConfigurationFromPath(configFilePath: string) {
         let fileData = fs.readFileSync(configFilePath, "utf8");
         // remove BOM if present
         fileData = fileData.replace(/^\uFEFF/, "");
-        return JSON.parse(fileData);
+        let config = JSON.parse(fileData) as IConfigFile;
+
+        let parentConfigNames = config.extends;
+        if (parentConfigNames != null) {
+            const configFileDir = path.dirname(configFilePath);
+            if (typeof parentConfigNames === "string") {
+                parentConfigNames = [<string>parentConfigNames];
+            }
+
+            for (const name of parentConfigNames) {
+                const parentConfigFilePath = resolveConfigFilePath(name, configFileDir);
+
+                let parentConfig = loadConfigurationFromPath(parentConfigFilePath);
+                config = applyExtends(config, parentConfig);
+            }
+        }
+
+        return config;
+    }
+}
+
+export interface IConfigFile {
+    extends: string | string[];
+    rulesDirectory: string | string[];
+    rules: any;
+}
+
+/**
+ * Resolve config file path.
+ * @var {string} relativeFilePath Relative path or
+ *   package name (tslint-config-X) or package short name (X)
+ */
+function resolveConfigFilePath(relativeFilePath: string, relativeTo: string) {
+    // `relativeFilePath.indexOf("/") === -1` will fail sub-rule use case when it is supported.
+    if (relativeFilePath.indexOf("/") === -1 && relativeFilePath.indexOf("tslint-config") === -1) {
+        relativeFilePath = `tslint-config-${relativeFilePath}`;
     }
+
+    if (relativeFilePath.indexOf("tslint-config-") === 0) {
+        let fileData = fs.readFileSync(path.resolve(`node_modules/${relativeFilePath}/package.json`), "utf8");
```

Comment:
Also, the original implementation only `require('<path to package.json>')` and get the tslintConfig, and manually read the file `using fs.readFileSync()` (and remove BOM). I don't know would `require('package-name')` would work correctly in this case.

Of course, I can change and test it. Have tests in place.